rust/private/rust.bzl

```diff
@@ -836,11 +836,36 @@ rust_library = rule(
         """),
 )
 
+def _rust_static_library_transition_impl(settings, attr):
```

Comment:
This looks good to me, but couldn't all these transitions be a single one shared by all rule types?

Reply:
An original version I did also had an optional `compilation_mode` transition that didn't apply to `rust_library()`, so I kept them different.  I've just moved them to using a `_change_platform()` call based on @katre's proposal.